Neos.Media/Classes/Domain/Service/ImageVariantService.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Neos\Media\Domain\Service;
+
+/*
+ * This file is part of the Neos.Media package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+use Neos\Media\Domain\ValueObject\Configuration\VariantPreset;
+use Neos\Flow\Annotations as Flow;
+
+/**
+ * An imageVariant service
+ *
+ * @Flow\Scope("singleton")
+ */
+class ImageVariantService
+{
+    /**
+     * Return all presets defined in 'Settings.Neos.Media.yaml' with presetName as key
+     *
+     * @param array $variantPresetConfigs
+     * @param string $presetIdentifier
+     * @return array
+     */
+    public function getAllPresetsByConfigs(array $variantPresetConfigs, string $presetIdentifier = ''): array
```

Comment:
Nitpicky, but we are commited to use as few shorts as possible, so I would prefer Configurations instead of Configs even if obivously in the context it's very clear what is meant here. Makes searching easier though.